app/frontend/src/wells/views/WellSearch.vue

```diff
@@ -228,8 +253,55 @@ export default {
         this.$store.commit(SET_SEARCH_RESULT_COLUMNS, query.result_columns.split(','))
         delete query.result_columns
       }
+      if (query.map_centre !== undefined) {
+        const latlng = query.map_centre.split(',')
+        const lat = parseFloat(latlng[0])
+        const lng = parseFloat(latlng[1])
+        this.$store.commit(SET_SEARCH_MAP_CENTRE, { lat, lng })
+        delete query.map_centre
+      }
+      if (query.map_zoom !== undefined) {
+        this.$store.commit(SET_SEARCH_MAP_ZOOM, parseInt(query.map_zoom))
+        delete query.map_zoom
+      }
+      if (query.constrain !== undefined) {
+        this.$store.commit(SET_CONSTRAIN_SEARCH, Boolean(query.constrain))
+        delete query.constrain
+      }
+
+      this.$store.commit(SET_SEARCH_PARAMS, query)
+    },
+    updateQueryParams () {
+      const tabHash = (this.tabIndex === 1) ? 'advanced' : null
 
-      this.$store.commit(SET_SEARCH_PARAMS, { ...query })
+      this.$router.replace({ query: this.buildQueryParams(), hash: tabHash })
+    },
+    buildQueryParams () {
```

Comment:
Would it make sense to have a unit test for this method?

Reply:
I have added a test that loads the component and triggers a `handleMapSearch` method on that component. That will force an update of the URL's query params calling `buildQueryParams`